brainscore_language/plugins/pereira2018/benchmark.py

```diff
@@ -0,0 +1,100 @@
+    recorded by Pereira et al. 2018.
+    Alignment of neural activity between model and human subjects is evaluated via cross-validated linear predictivity.
+
+    This benchmark builds off the behavioral benchmark introduced
+    in Schrimpf et al. 2021 (https://www.pnas.org/doi/10.1073/pnas.2105646118), but:
+
+    * computes neural alignment to each of the two experiments ({243,384}sentences) separately, as well as ceilings
+    * requires the model to have committed to neural readouts (e.g. layer 41 corresponds to the language system),
+        rather than testing every layer separately
+
+    Each of these benchmarks evaluates one of the two experiments, the overall Pereira2018-linear score is the mean of
+    the two ceiling-normalized scores.
+    """
+
+    def __init__(self, experiment: str, ceiling_s3_kwargs: dict):
+        self.data = self._load_data(experiment)
+        self.metric = load_metric('linear_pearsonr')
+        identifier = f'Pereira2018.{experiment}-linear'
+        ceiling = self._load_ceiling(identifier=identifier, **ceiling_s3_kwargs)
+        super(_Pereira2018ExperimentLinear, self).__init__(
+            identifier=identifier,
+            version=1,
+            parent='Pereira2018-linear',
+            ceiling=ceiling,
+            bibtex=BIBTEX)
+
+    def _load_data(self, experiment: str) -> NeuroidAssembly:
+        data = load_dataset('Pereira2018.language')
+        data = data.sel(experiment=experiment)  # filter experiment
+        data = data.dropna('neuroid')  # not all subjects have done both experiments, drop those that haven't
+        data.attrs['identifier'] = f"{data.identifier}.{experiment}"
+        return data
+
+    def _load_ceiling(self, identifier: str, version_id: str, sha1: str, assembly_prefix="ceiling_", raw_kwargs=None):
+        ceiling = load_from_s3(identifier, cls=Score, assembly_prefix=assembly_prefix, version_id=version_id, sha1=sha1)
+        if raw_kwargs:  # recursively load raw attributes
+            raw = self._load_ceiling(identifier=identifier, assembly_prefix=assembly_prefix + "raw_", **raw_kwargs)
+            ceiling.attrs['raw'] = raw
+        return ceiling
+
+    def __call__(self, candidate: ArtificialSubject) -> Score:
+        candidate.perform_neural_recording(recording_target=ArtificialSubject.RecordingTarget.language_system,
+                                           recording_type=ArtificialSubject.RecordingType.fMRI)
+        stimuli = self.data['stimulus']
+        predictions = candidate.digest_text(stimuli.values)['neural']
+        predictions['stimulus_id'] = 'presentation', stimuli['stimulus_id'].values
+        raw_score = self.metric(predictions, self.data)
+        score = ceiling_normalize(raw_score, self.ceiling)
+        return score
+
+
+benchmarks['Pereira2018.243sentences-linear'] = Pereira2018_243sentences
+benchmarks['Pereira2018.384sentences-linear'] = Pereira2018_384sentences
```

Comment:
is it a little dangerous to have a central object `benchmarks` that is modified across all submodules? what if there was a central Registry instance with methods to add plugins to itself using something like `.register_benchmark(identifier: typing.str, cls: typing.Callable)`

Reply:
plugin system by @kvfairchild is planned to take care of this